src/components/clusterDetail/FailedHostsWarning.tsx

```diff
@@ -1,38 +1,68 @@
 import React from 'react';
 import { GridItem, Alert, AlertVariant, TextContent, Text } from '@patternfly/react-core';
-import { Cluster } from '../../api/types';
+import { Cluster, Host } from '../../api/types';
 
 type FailedHostsWarningProps = {
   cluster: Cluster;
 };
 
-const FailedHostsWarning: React.FC<FailedHostsWarningProps> = ({ cluster }) => {
-  const failedHosts =
-    cluster.hosts?.filter((host) => host.role === 'worker' && host.status === 'error') || [];
+type FailedHostsProps = {
+  failedHosts: Host[];
+  title: string;
+  titleSingle: string;
+  text: string;
+};
 
+const FailedHosts: React.FC<FailedHostsProps> = ({ failedHosts, title, titleSingle, text }) => {
   if (failedHosts.length === 0) {
     return null;
   }
 
-  let title;
-  if (failedHosts.length === 1) {
-    title = `1 worker host failed to install`;
-  } else {
-    title = `${failedHosts.length} worker hosts failed to install`;
-  }
-
   return (
     <GridItem>
-      <Alert variant={AlertVariant.warning} title={title} isInline>
+      <Alert
+        variant={AlertVariant.warning}
+        title={failedHosts.length === 1 ? titleSingle : title}
+        isInline
+      >
         <TextContent>
-          <Text component="p">
-            Error information for each worker host is available below. Debug and reboot each host to
-            try again.
-          </Text>
+          <Text component="p">{text}</Text>
         </TextContent>
       </Alert>
     </GridItem>
   );
 };
 
+const FailedHostsWarning: React.FC<FailedHostsWarningProps> = ({ cluster }) => {
+  const failedHosts =
+    cluster.hosts?.filter((host) => host.role === 'worker' && host.status === 'error') || [];
+
+  const failedBootOrder =
+    cluster.hosts?.filter((host) => host.status === 'installing-pending-user-action') || [];
+
+  let failedBootOrderText =
+    "Please reconfigure each host's BIOS to boot from the disk where OpenShift was installed instead of the Discovery ISO.";
+  if (failedBootOrder.find((host) => host.role === 'master')) {
+    failedBootOrderText +=
+      ' The cluster installation will time-out after 30 minutes if a master node does not reboot properly.';
```

Comment:
```suggestion
      ' The cluster installation will time-out after 30 minutes if a master host does not reboot properly.';
```

Aligns better with the `1 worker host` message below. I think calling them "hosts" in the installer in general probably makes sense, since "nodes" are kind of what they turn into after the cluster is up.

Reply:
done